design/README.md

```diff
@@ -0,0 +1,15 @@
+# ROS2 Navigation Design documenation
+This is where the ROS2 Navigation design documentation is being collected and vetted
```

Comment:
"ROS2 Navigation" here is inconsistent with the previous README.md, which uses "ROS2 navigation"